### PR TITLE
Update upgrade script for the new layout of secrets.

### PR DESCRIPTION
**Pull request by andreas-kupries**

### CODE REVIEW

File: make/upgrade [hunk 1]
```diff
@@ -8,16 +8,24 @@ set -o errexit -o nounset
...
 NAMESPACE="cf"
 UAA_NAMESPACE="uaa"
-UAA_CA_CERT=internal-ca-cert
 
 GIT_ROOT=${GIT_ROOT:-$(git rev-parse --show-toplevel)}
 cd "${GIT_ROOT}"
 
-# Determine the kube revision of the chart controlling the specified namespace.
-release_version() { helm list --date --reverse --max 1 --namespace "$1" | awk '{ print $2 }' | tail -n 1 ; }
+get_uaa_secret_name() {
```
Comment (jandubois) [inline make/upgrade]: I think this function should take an argument for `INTERNAL_CA_CERT` and not have it hard-coded.

File: make/upgrade [hunk 2]
```diff
@@ -8,16 +8,24 @@ set -o errexit -o nounset
...
+    kubectl get pods \
+	--namespace "${UAA_NAMESPACE}" \
+	-o jsonpath='{.items[*].spec.containers[?(.name=="uaa")].env[?(.name=="INTERNAL_CA_CERT")].valueFrom.secretKeyRef.name}'
+}
+
+get_uaa_secret_key() {
```
Comment (jandubois) [inline make/upgrade]: Same here

File: make/upgrade [hunk 3]
```diff
@@ -8,16 +8,24 @@ set -o errexit -o nounset
...
+    kubectl get pods \
+	--namespace "${UAA_NAMESPACE}" \
+	-o jsonpath='{.items[*].spec.containers[?(.name=="uaa")].env[?(.name=="INTERNAL_CA_CERT")].valueFrom.secretKeyRef.key}'
+}
 
 get_uaa_secret () {
```
Comment (jandubois) [inline make/upgrade]: This function also should take the env variable name as an argument, and not the name of the secret key. Instead it should call `$(get_uaa_secret_key $ENV_NAME)` itself.

File: make/upgrade [hunk 4]
```diff
@@ -8,16 +8,24 @@ set -o errexit -o nounset
...
-    kubectl get secret secret-$(release_version "${UAA_NAMESPACE}") \
+    kubectl get secret $(get_uaa_secret_name) \
```
Comment (jandubois) [inline make/upgrade]: This call should move inside `get_uaa_secret`.

File: make/upgrade [hunk 5]
```diff
@@ -44,7 +52,7 @@ fi
 source bin/settings/settings.env
 source "${NETWORK_ENV}"
 
-UAA_CA_CERT="$(get_uaa_secret ${UAA_CA_CERT} | base64 -d -)"
+UAA_CA_CERT="$(get_uaa_secret $(get_uaa_secret_key) | base64 -d -)"
```
Comment (jandubois) [inline make/upgrade]: The argument here should be `INTERNAL_CA_CERT`

File: make/upgrade (hunk 4 above)
Comment (jandubois) [reply make/upgrade]: Hmm, this comment looks like it got attached to the wrong line

File: make/upgrade (hunk 1 above)
Comment (andreas-kupries) [reply make/upgrade]: Done with commit 405a6bdb262d41482dfb7d8e5045cf3f406aa183

File: make/upgrade (hunk 2 above)
Comment (andreas-kupries) [reply make/upgrade]: Done with commit 405a6bdb262d41482dfb7d8e5045cf3f406aa183

File: make/upgrade (hunk 3 above)
Comment (andreas-kupries) [reply make/upgrade]: Done with commit 405a6bdb262d41482dfb7d8e5045cf3f406aa183

File: make/upgrade (hunk 4 above)
Comment (andreas-kupries) [reply make/upgrade]: Done with commit 405a6bdb262d41482dfb7d8e5045cf3f406aa183

File: make/upgrade (hunk 5 above)
Comment (andreas-kupries) [reply make/upgrade]: Done with commit 405a6bdb262d41482dfb7d8e5045cf3f406aa183

**Pull request merged**